### PR TITLE
fix: add data to snapshot

### PR DESCRIPTION
**Pull request by Caele**
The snapshot now only stores the data it needs, which is the nodes you see, plus their children (to get the numbers right, bit of a shortcut).

### CODE REVIEW

File: src/tree/render.js [hunk 1]
```diff
@@ -4,17 +4,24 @@ import box from './box';
...
-      if (child.children && expandedChildren.indexOf(child.data.id) !== -1) {
-        child.children.forEach(grandChild => {
-          subTree.push(grandChild);
-        });
+      if (child.children) {
+        if (expandedChildren.indexOf(child.data.id) !== -1 || extended) {
+          child.children.forEach(grandChild => {
+            subTree.push(grandChild);
+            if (expandedChildren.indexOf(child.data.id) !== -1 && extended && grandChild.children) {
+              child.children.forEach(extendedChild => {
```
Comment (haxxmaxx) [inline src/tree/render.js]: Doesn't this just att the granchild twice?

Comment (niekvanstaveren) [reply src/tree/render.js]: Yes this should be grandChild.children.forEach I think

Comment (Caele) [reply src/tree/render.js]: yea it should, how the heck did this work then, or seem to work.

Comment (Caele) [reply src/tree/render.js]: I did not look closely at that =)

File: src/index.js [hunk 2]
```diff
@@ -155,7 +154,7 @@ export default function supernova(env) {
        * - render [allNodes, positioning, styling, rect]
        */
 
-      usePromise(() => {
+      const [dataTree] = usePromise(() => {
```
Comment (Caele) [inline src/index.js]: This was likely not needed, but removes a state for us.

File: src/utils/tree-utils.js [hunk 3]
```diff
@@ -181,11 +186,11 @@ export function createNodes(matrix, attributeIndecies, status, navigationMode, t
     const parentNode = nodeMap[node.parentId];
     node.parent = parentNode;
     if (parentNode) {
-      parentNode.children.length > 100
+      parentNode.children.length > 98
```
Comment (Caele) [inline src/utils/tree-utils.js]: Changed this to 98, so that the max number of children becomes 99, which fits in the +/- box.

**Pull request merged**